newIDE/app/src/EventsSheet/EventsTree/Instruction.js

```diff
@@ -289,17 +302,38 @@ const Instruction = (props: Props) => {
                 .getRootNode();
               const expressionValidator = new gd.ExpressionValidator(
                 gd.JsPlatform.get(),
-                getProjectScopedContainersFromScope(
-                  scope,
-                  globalObjectsContainer,
-                  objectsContainer
-                ),
+                projectScopedContainers,
                 parameterType
               );
               expressionNode.visit(expressionValidator);
               expressionIsValid =
                 expressionValidator.getAllErrors().size() === 0;
               expressionValidator.delete();
+
+              // New object variable instructions require the variable to be
+              // declared while legacy ones don't.
+              // This is why it's done here instead of in the parser directly.
+              if (
+                parameterType === 'objectvar' &&
+                gd.VariableInstructionSwitcher.isSwitchableVariableInstruction(
+                  instruction.getType()
+                )
```

Comment:
For global and scene variable, there is a new parameter type so we don't have this issue.

Reply:
Ok! 👍